### ISSUE TITLE
Don't override MySql 'localhost' with /tmp/mysql.sock for Docker

### ISSUE THREAD

**Issue by cuyler** [TablePlus/TablePlus#1169]
1. Which driver are you using and version of it (Ex: PostgreSQL 10.0):

MySQL 5.x

2. Which TablePlus build number are you using (the number on the welcome screen, Ex: build 81):

222

3. The steps to reproduce this issue:

Create a new connection, fallout connection information, set the "Host" to 'localhost' then watch the connection fail, because there's no /tmp/mysql.sock.

I think it's trying to be 'helpful' but when running a docker instance of a database, localhost:[port] is much easier to reason about ( especially when you're DHCP ) than having to use the currently leased IP address.

In short, I think respecting the "use socket" check-box would be more helpful than not.

**Comment by huyphams** [comment]
Thank @cuyler but TablePlus always respect the input. I don’t know if the driver handles it automatically. Please try to use 127.0.0.1 instead.

**Comment by huyphams** [comment]
I checked the MySQL document, it is handled automatically:


> On Unix, MySQL programs treat the host name localhost specially, in a way that is likely different from what you expect compared to other network-based programs.
> 
> Clients determine what type of connection to make as follows:
> 
> If the host is not specified or is localhost, a connection to the local host is assumed:
> 
> On Windows, the client connects using a shared-memory connection, if the server has shared-memory connections enabled.
> 
> On Unix, the client connects using a Unix socket file. The --socket option or the MYSQL_UNIX_PORT environment variable may be used to specify the socket name.

I think I will not change the default behavior because it's the standard of MySQL. But I will add an alert to tell the user that they should change to the 127.0.0.1 if there is an error.

https://dev.mysql.com/doc/refman/8.0/en/connecting.html

**Comment by huyphams** [comment]
I updated the popup to clarify it.
<img width="612" alt="Screen Shot 2019-05-07 at 8 27 33 AM" src="https://user-images.githubusercontent.com/3654622/57265449-fc830000-70a1-11e9-9082-354f068f1b98.png">

**Comment by huyphams** [comment]
I've added it on this build: https://www.dropbox.com/s/0i0ofnhi88vlxz8/TablePlus.dmg?dl=0
Thank you for the feedback.
